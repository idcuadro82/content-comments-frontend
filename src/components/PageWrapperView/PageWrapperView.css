@import '../../css/colors.css';

.document {
  background: var(--light-gray);
  display: flex;
  justify-content: center;
  width: 100%;
  height: 100%;
}

.document-content {
  background-color: var(--white);
  height: fit-content;
  max-width: 1024px;
  margin-top: 40px;
  padding: 32px 24px;
  width: 100%;
}
