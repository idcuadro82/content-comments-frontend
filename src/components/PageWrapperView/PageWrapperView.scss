@import 'src/styles/colors';

.document {
  align-items: flex-start;
  background: $light-gray;
  display: flex;
  justify-content: center;
  width: 100%;
  height: 100%;

  &-content {
    background-color: $white;
    max-width: 1024px;
    margin-top: 40px;
    padding: 32px 24px;
    width: 100%;

    .document-block-list {
      display: flex;
      flex-direction: column;
    }
  }
}
