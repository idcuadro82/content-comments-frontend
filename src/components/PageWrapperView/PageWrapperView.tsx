import React, { FunctionComponent } from 'react';
import PropTypes from 'prop-types';
import './PageWrapperView.css';

const PageWrapperView: FunctionComponent = ({ children }) => {
  return (
    <div className="document">
      <div className="document-content">{children}</div>
    </div>
  );
};

PageWrapperView.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
};

export default PageWrapperView;
