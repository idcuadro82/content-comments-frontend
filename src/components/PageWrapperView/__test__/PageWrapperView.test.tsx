import React from 'react';
import PageWrapperView from '../PageWrapperView';
import { render } from '@testing-library/react';


describe('<PageWrapperView>', () => {

  it('Render <PageWrapperView>', () => {
    const { container, queryByTestId } = render(
      <PageWrapperView>
        <h1 data-testid="page-wrapper">Book 1</h1>
      </PageWrapperView>
    );

    expect(queryByTestId('page-wrapper')).toBeTruthy();
    expect(container).toMatchSnapshot();
  })
});
