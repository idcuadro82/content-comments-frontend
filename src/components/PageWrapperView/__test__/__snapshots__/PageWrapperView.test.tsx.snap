// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageWrapperView> Render <PageWrapperView> 1`] = `
<div>
  <div
    class="document"
  >
    <div
      class="document-content"
    >
      <h1
        data-testid="page-wrapper"
      >
        Book 1
      </h1>
    </div>
  </div>
</div>
`;
